runtime/lua/vim/lsp/buf.lua

```diff
@@ -134,14 +170,58 @@ end
 ---
 --@param options Table with valid `FormattingOptions` entries
 --@param timeout_ms (number) Request timeout
+--@see |vim.lsp.buf.formatting_seq_sync|
 function M.formatting_sync(options, timeout_ms)
+  local client = get_formatting_client()
+  if client == nil then return end
+
   local params = util.make_formatting_params(options)
-  local result = vim.lsp.buf_request_sync(0, "textDocument/formatting", params, timeout_ms)
-  if not result or vim.tbl_isempty(result) then return end
-  local _, formatting_result = next(result)
-  result = formatting_result.result
-  if not result then return end
-  vim.lsp.util.apply_text_edits(result)
+  local result = client.request_sync("textDocument/formatting", params, timeout_ms)
+  if result and result.result then
+    util.apply_text_edits(result.result)
+  end
```

Comment:
Should we also give the users some feedback if there is an error. Maybe using `vim.notify` with the error message?

Reply:
Makes sense to me. Probably a good idea for any API function that has a `timeout_ms` parameter.

Reply:
Done